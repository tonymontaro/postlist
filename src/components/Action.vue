<template>
  <li :class="boxStype">
    {{ action.word }}
    <button :class="btnStyle" @click="() => replyHistory(action.prevState)">
      Time travel
    </button>
  </li>
</template>

<script>
import { BOX_STYLE } from "@/config";

export default {
  name: "action",
  props: ["action", "replyHistory"],
  data() {
    return {
      boxStype: [...BOX_STYLE],
      btnStyle: [
        "bg-green-200",
        "hover:bg-green-300",
        "py-1",
        "px-2",
        "rounded",
        "ml-3"
      ]
    };
  }
};
</script>
