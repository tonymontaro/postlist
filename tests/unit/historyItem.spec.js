import { mount } from "@vue/test-utils";

import Action from "@/components/HistoryItem";

describe("HistoryItem Component", () => {
  test("renders post item with up and down links", () => {
    const wrapper = mount(Action, {
      propsData: {
        historyItem: {
          word: "Moved post 1 from index 1 to index 2",
          prevState: []
        },
        replyHistory: () => {}
      }
    });

    expect(wrapper.findAll("button")).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
