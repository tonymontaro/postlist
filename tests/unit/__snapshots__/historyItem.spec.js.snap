// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem Component renders post item with up and down links 1`] = `
<li class="p-3 py-5 bg-white shadow-lg rounded-sm white-box w-full">
  Moved post 1 from index 1 to index 2
  <button class="bg-green-200 hover:bg-green-300 py-1 px-2 rounded ml-3">
    Time travel
  </button></li>
`;
